src/WingetCreateCLI/Commands/NewCommand.cs

```diff
@@ -237,10 +239,89 @@ private static void PromptRequiredProperties<T>(T manifest, VersionManifest vers
+        }
+
+        /// <summary>
+        /// Displays all installers from an Installer manifest as a selection menu.
+        /// </summary>
+        private static void DisplayInstallersAsMenuSelection(InstallerManifest installerManifest)
+        {
+            Console.Clear();
+
+            while (true)
+            {
+                List<string> selectionList = GenerateInstallerSelectionList(installerManifest.Installers, out Dictionary<string, Installer> installerSelectionMap);
+                var selectedItem = Prompt.Select(Resources.SelectInstallerToEdit_Message, selectionList);
+
+                if (selectedItem == Resources.None_MenuItem)
+                {
+                    break;
+                }
+                else if (selectedItem == Resources.AllInstallers_MenuItem)
+                {
+                    Installer installerCopy = new Installer();
+                    PromptHelper.PromptPropertiesWithMenu(installerCopy, Resources.None_MenuItem);
+                    ApplyChangesToIndividualInstallers(installerCopy, installerManifest.Installers);
+                }
+                else if (selectedItem == Resources.DisplayPreview_MenuItem)
+                {
+                    Console.Clear();
+                    Console.WriteLine();
+                    Logger.InfoLocalized(nameof(Resources.DisplayPreviewOfSelectedInstaller_Message));
+                    var serializer = Serialization.CreateSerializer();
+                    string installerString = serializer.Serialize(installerManifest);
+                    Console.WriteLine(installerString);
+                    Console.WriteLine();
+                }
+                else
+                {
+                    Installer selectedInstaller = installerSelectionMap[selectedItem];
+                    PromptHelper.PromptPropertiesWithMenu(selectedInstaller, Resources.None_MenuItem);
+                }
+            }
+        }
+
+        private static List<string> GenerateInstallerSelectionList(List<Installer> installers, out Dictionary<string, Installer> installerSelectionMap)
+        {
+            installerSelectionMap = new Dictionary<string, Installer>();
+
+            foreach (Installer installer in installers)
+            {
+                var installerTuple = string.Join(" | ", new[]
+                    {
+                        installer.Architecture.ToEnumAttributeValue(),
+                        installer.InstallerType.ToEnumAttributeValue(),
+                        installer.Scope?.ToEnumAttributeValue(),
+                        installer.InstallerLocale,
+                        installer.InstallerUrl,
+                    }.Where(s => !string.IsNullOrEmpty(s)));
+
+                installerSelectionMap.Add(installerTuple, installer);
+            }
+
+            List<string> selectionList = new List<string>() { Resources.AllInstallers_MenuItem };
+            selectionList.AddRange(new List<string>(installerSelectionMap.Keys));

```

Comment:
nit: You don't need to create a new list here in the AddRange

Reply:
removed new list from addrange